rust/src/array.rs

```diff
@@ -626,6 +740,88 @@ mod tests {
         PrimitiveArray::<i32>::from(data);
     }
 
+    #[test]
+    fn test_boolean_array_new() {
+        // 00000010 01001000
+        let buf = Buffer::from([72_u8, 2_u8]);
+        let buf2 = buf.clone();
+        let pa = BooleanArray::new(10, buf, 0, 0);
```

Comment:
Just a note, that `pa` naming confuses me a little, because the python impl uses `import pyarrow as pa` :)